rts/Lua/LuaUnsyncedCtrl.cpp

```diff
@@ -3541,6 +3543,91 @@ int LuaUnsyncedCtrl::SetBuildFacing(lua_State* L)
 }
 
 
+// FIXME: move to LuaUtils, but currently it produces error
+// "error: undefined reference to 'BuildInfo::BuildInfo(int, float3 const&, int)'"
+static BuildInfo ParseBuildInfo(
+	lua_State* L,
+	const char* caller,
+	int idx
+) {
+	int unitDefID = luaL_checkinteger(L, idx);
+	float3 pos = {
+		luaL_checkfloat(L, idx+1),
+		luaL_checkfloat(L, idx+2),
+		luaL_checkfloat(L, idx+3),
+	};
+	int facing = luaL_checkinteger(L, idx+4);
+
+	return BuildInfo(unitDefID, pos, facing);
+}
+
+
+static LuaBuildSquareOptions ParseBuildSquareOptions(
+	lua_State* L,
+	const char* caller,
+	const int idx
+) {
+	LuaBuildSquareOptions opts;
+	if (lua_istable(L, idx)) {
+		for (lua_pushnil(L); lua_next(L, idx) != 0; lua_pop(L, 1)) {
+			if (lua_israwstring(L, -2)) {
+				switch (hashString(lua_tostring(L, -2))) {
+					case hashString("unbuildable"):
+						opts.unbuildable = luaL_checkboolean(L, -1);
+						break;
+					case hashString("cacheValidity"):
+						opts.cacheValidity = int(luaL_checkfloat(L, -1) * GAME_SPEED);
+						break;
+				}
+			}
+		}
+	} else if (!lua_isnone(L, idx) && !lua_isnil(L, idx)) {
+		luaL_error(L, "%s(): bad options-argument type", caller);
+	}
+	return opts;
+}
+
+
+/***
+ *
+ * @function Spring.AddUnitBuildSquare
+ * @number unitDefID
+ * @number x
+ * @number y
+ * @number z
+ * @number facing
+ * @bool[opt] unbuildable
```

Comment:
Seems to be a table now.

Reply:
I've tried to fix those comments, I'm still not clear how they work though because they are quite different from LUA annotations.